src/solvers/strings/string_dependencies.h

```diff
@@ -0,0 +1,200 @@
+/*******************************************************************\
+
+Module: String solver
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#ifndef CPROVER_SOLVERS_STRINGS_STRING_DEPENDENCIES_H
+#define CPROVER_SOLVERS_STRINGS_STRING_DEPENDENCIES_H
+
+#include <memory>
+
+#include "string_builtin_function.h"
+#include "string_constraint.h"
+#include "string_constraint_generator.h"
+
+/// Keep track of dependencies between strings.
+/// Each string points to the builtin_function calls on which it depends.
+/// Each builtin_function points to the strings on which the result depends.
```

Comment:
:bulb: I think this documentation should answer the question "What do we mean by `dependency` and how do we use dependencies?". An example, or reference to markdown files about the string solver with examples, would be helpful here. I don't know much about the string solver so I'm mostly guessing what any of this might mean. :slightly_smiling_face: